endpoint/endpoint.go

```diff
@@ -145,3 +154,37 @@ func NewEndpointWithTTL(dnsName, recordType string, ttl TTL, targets ...string)
 func (e *Endpoint) String() string {
 	return fmt.Sprintf("%s %d IN %s %s", e.DNSName, e.RecordTTL, e.RecordType, e.Targets)
 }
+
+// DNSEndpointSpec defines the desired state of DNSEndpoint
+type DNSEndpointSpec struct {
+	Endpoints []*Endpoint `json:"endpoints,omitempty"`
+}
+
+// DNSEndpointStatus defines the observed state of DNSEndpoint
+type DNSEndpointStatus struct {
+	// The generation observed by the external-dns controller.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DNSEndpoint is the CRD wrapper for Endpoint
```

Comment:
I would suggest something like:

```
// DNSEndpoint is a contract that a user-specified CRD must implement to be used as a source for external-dns.
```

Reply:
I would also call out that the user-specified CRD must have the status subresource.

Reply:
Updated.